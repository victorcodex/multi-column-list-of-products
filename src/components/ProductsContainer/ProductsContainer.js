import styled from "styled-components";

const ProductsContainer = styled.section`
  display: flex;
  flex-wrap: wrap;
  & figure.product-box-item {
    margin-bottom: 5vh;
    width: calc(50% - 1vw);
    text-align: center;
    margin: 0;
  }
  & figure.product-box-item figcaption  {
    margin-top: 2vh;
  }
  & figure.product-box-item figcaption.product-name  {
    padding: 0 3vh;
  }
  & figure.product-box-item figcaption.product-type  {
    color: #737373;
  }
  & figure.product-box-item figcaption.product-price-wrapper  {
    font-size: 1rem;
    line-height: 1.2;
  }
  & figure.product-box-item img  {
    height: auto;
    max-width: 100%;
  }
  @media screen and (min-width: 800px) {
    figure.product-box-item {
      width: calc(40% - 6vh);
    }
  }
`;

export default ProductsContainer;