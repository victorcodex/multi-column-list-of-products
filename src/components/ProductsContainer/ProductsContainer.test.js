import React from 'react';
import { render } from '@testing-library/react';
import ProductsContainer from './ProductsContainer';

describe('ProductsContainer component', () => {
  it('should render product container', () => {
    const { asFragment } = render(<ProductsContainer rating={100}></ProductsContainer>);
    expect(asFragment()).toMatchSnapshot();
  });
});