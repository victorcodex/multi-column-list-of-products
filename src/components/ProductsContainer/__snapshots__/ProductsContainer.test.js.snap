// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsContainer component should render product container 1`] = `
<DocumentFragment>
  <section
    class="sc-bdnxRM gyxSxt"
  />
</DocumentFragment>
`;
